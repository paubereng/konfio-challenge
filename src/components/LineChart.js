import React from 'react';
import PropTypes from 'prop-types';
import {
  LineChart, Line, XAxis, YAxis, Tooltip, Label, Legend, ResponsiveContainer,
} from 'recharts';
import omit from 'lodash/omit';
import { COLORS } from '../constants';

const INITIAL_DATA = [
  {
    name: 'EUR', EUR: 0, USD: 0, GBP: 0,
  },
];

const customLabelLegend = name => `Date: ${name}`;

const SimpleLineChart = ({ rates }) => {
  const data = rates && rates.length > 0 ? rates : INITIAL_DATA;
  const currencies = Object.keys(omit(data[0], 'name'));
  return (
    <ResponsiveContainer width="99%" height="80%">
      <LineChart width={ 600 } height={ 300 } data={ data }
      margin={
        {
          top: 5, right: 25, bottom: 45, left: 5,
        }
      }>
        <XAxis dataKey="name">
          <Label value="Dates" offset={-20} position="insideBottom" />
        </XAxis>
        <YAxis>
          <Label value='Rates' angle={-90} position='insideLeft' />
        </YAxis>
        <Tooltip labelFormatter={customLabelLegend} />
        <Legend
          wrapperStyle={{
            paddingTop: 40, paddingLeft: 50,
          }}
        />
        { currencies.map((currency, index) => <Line key={ index } strokeWidth={2} type="monotone" stroke={COLORS[index]} dataKey={ currency } />) }
      </LineChart>
    </ResponsiveContainer>
  );
};

SimpleLineChart.propTypes = {
  rates: PropTypes.array,
};

export default SimpleLineChart;
