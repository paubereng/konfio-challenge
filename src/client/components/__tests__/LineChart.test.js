import React from 'react';
import { render } from 'enzyme';
import LineChart from '../LineChart';

describe('LineChart component', () => {
  let component;
  const mockProps = {
    rates: [
      { name: '2019-04-26', EUR: 0.8, USD: 0.5, GBP: 0.6 },
      { name: '2019-04-27', EUR: 0.9, USD: 0.4, GBP: 0.8 },
    ],
  }

  beforeEach(()=>{
    component = render(<LineChart { ...mockProps } />);
  });

  it('renders correctly', () => {
      expect(component).toMatchSnapshot();
  });
});
