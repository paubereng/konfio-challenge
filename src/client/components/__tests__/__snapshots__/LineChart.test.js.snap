// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart component renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "recharts-responsive-container",
      "style": "width:99%;height:500px",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "style": "position:absolute;width:0;height:0;visibility:hidden;display:none",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
